.github/workflows/test-apt-packages.yml

```diff
@@ -0,0 +1,82 @@
+name: Run smoke tests on APT packages
+
+on:
+  workflow_call:
+    inputs:
+      pkg_type:
+        required: true
+        type: string
+      revision_name:
+        required: true
+        type: string
+
+jobs:
+  TestPackagesWith:
+    strategy:
+      matrix:
+        distro:
+          - "debian:stretch"
+          - "debian:buster"
+          - "debian:bullseye"
+          - "debian:testing"
+          - "debian:sid"
+          - "ubuntu:xenial"
+          - "ubuntu:bionic"
+          - "ubuntu:focal"
+          - "ubuntu:impish"
+      fail-fast: false
+    runs-on: ubuntu-latest
+    container: ${{ matrix.distro }}
+    steps:
+      - name: Install prerequisites for adding 3rd party repository
+        run: |
+          apt-get update -qq
+          apt-get install --yes wget gnupg2 ca-certificates apt-transport-https
+
+      - name: Add nightly OSE repository
+        if: inputs.pkg_type == 'nightly'
+        run: |
+          wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | gpg --dearmor > /usr/share/keyrings/ose-repo-archive-keyring.gpg
+          echo "deb [trusted=yes] https://ose-repo.syslog-ng.com/apt/ ${{ inputs.pkg_type }} $(echo ${{ matrix.distro }} | sed 's/:/-/g')" | tee --append /etc/apt/sources.list.d/syslog-ng-ose.list
+
+      - name: Add stable OSE repository
+        if: inputs.pkg_type == 'stable'
+        run: |
+          wget -qO - https://ose-repo.syslog-ng.com/apt/syslog-ng-ose-pub.asc | gpg --dearmor > /usr/share/keyrings/ose-repo-archive-keyring.gpg
+          echo "deb [signed-by=/usr/share/keyrings/ose-repo-archive-keyring.gpg] https://ose-repo.syslog-ng.com/apt/ ${{ inputs.pkg_type }} $(echo ${{ matrix.distro }} | sed 's/:/-/g')" | tee --append /etc/apt/sources.list.d/syslog-ng-ose.list
+
+      - name: Install the last but one syslog-ng OSE package
+        run: |
+          apt-get update -qq
+          DEBIAN_FRONTEND=noninteractive apt-get install --yes syslog-ng=$(apt-cache madison syslog-ng | sed -n 2p | awk -F"|" '{print $2}' | sed 's/ //g')
+
+      - name: Upgrade to the latest syslog-ng OSE package
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt-get install --yes syslog-ng
+
+      - name: Get syslog-ng revision
+        run: |
+          syslog-ng -V
+          echo "::set-output name=REVISION::$(syslog-ng -V | grep Revision | cut -d" " -f2)"
+        id: syslog_ng_revision
+
+      - name: Check if revision is snapshot revision
+        if: inputs.revision_name == 'snapshot'
+        run: |
+          echo ${{ steps.syslog_ng_revision.outputs.REVISION }} | grep ${{ inputs.revision_name }}
```

Comment:
Do we need this input field? We can use the same `inputs.pkg_type` variable here too. `nightly` would mean check for `"snapshot"`.

Alternatively we can keep the`revision_name` input variable, and use just one revision-checking step, where the input itself is the pattern to check for: e.g. `snapshot` or `^[0-9]{1}.[0-9]{1,2}.[0-9]{1}-[0-9]{1}$`.

Reply:
Thanks, I like your idea.
What if we drop the "revision_name" input variable and also we will check the revision in one step.
Something like this:
```
      - name: Check revision
        if: inputs.pkg_type == 'nightly'
        run: |
          echo ${{ steps.syslog_ng_revision.outputs.REVISION }} | egrep "^[0-9]{1}.[0-9]{1,2}.[0-9]{1}.[0-9]{3}.[a-z0-9]{8}-snapshot+[0-9]{8}T[0-9]{6}$"
        if: inputs.pkg_type == 'stable'
        run: |
          echo ${{ steps.syslog_ng_revision.outputs.REVISION }} | egrep "^[0-9]{1}.[0-9]{1,2}.[0-9]{1}-[0-9]{1}$"
```

Reply:
My first idea was to link logically the different (snapshot and stable) revision checks to the steps as it is linked in case of installing different packages.
But this new revision_name input field has almost a redundant meaning with the pkg_type value (which represents how we used them in real).
It would be clearer if they has the same usage in real: in apt source and in installer revision.

Reply:
Removing the unnecessary `revision_name` input variable sounds good to me.
I'm not sure that your first example would work with multiple `run` and `if` statements, but you can merge the `run` statements, and you can use an `if` statement on bash level, i.e. inside the `run` statement:
```
run: |
  if [[ ${{ inputs.pkg_type }} == "nightly" ]]; then
    echo "${{ steps.syslog_ng_revision.outputs.REVISION }}" | egrep "^[0-9]{1}.[0-9]{1,2}.[0-9]{1}.[0-9]{3}.[a-z0-9]{8}-snapshot+[0-9]{8}T[0-9]{6}$";
  fi
  elif [[ ${{ inputs.pkg_type }} == "stable" ]]; then
    echo "${{ steps.syslog_ng_revision.outputs.REVISION }}" | egrep "^[0-9]{1}.[0-9]{1,2}.[0-9]{1}-[0-9]{1}$"
  fi
```